src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -145,6 +148,61 @@ public void to(MutationComposer composer) throws CommitConflictException {
             });
   }
 
+  private boolean isWriteSetOverlappedWith(Scan scan) {
```

Comment:
I was wondering if we need to consider `limit` and `projections` of the scan operation here. But, I found we clear `projections` at the following line, so it looks like we don't need to consider `projections`:
https://github.com/scalar-labs/scalardb/blob/master/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommit.java#L92

I have two questions:
- Why do we clear `projections` in scan/get operations in `ConsensusCommit`?
- Do we need to consider `limit` in `isWriteSetOverlappedWith`?

Thanks.

Reply:
Thank you for the questions!

> Why do we clear projections in scan/get operations in ConsensusCommit?

This is because ConsensusCommit stores its metadata in the columns of a record and needs to force applications to get all the columns to properly handle transactions.
Even if it doesn't use such metadata columns, as long as conflicts are detected on the basis of records (not columns), projections shouldn't be taken into consideration.

> Do we need to consider limit in isWriteSetOverlappedWith?

I don't think so.
First, `limit` cannot be utilized here because there is no way to identify what records would be returned by the limit.
Even if we know what records would be returned by the `limit`, 
the same issue still exists if the records and write set don't overlap.
Essentially, conflict detection for scan should be done with a specified range (like predicate locking) instead of returned records otherwise it would cause phantom.

But, I might have missed your points, so please feel free to ask more questions and concerns.

Reply:
Thank you!